### PR TITLE
Rename Family to Case

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR renames the `Family` model to `Case`. This is something we want to do, right? I figure the tables, other models, fields in other models, functions and yada yada can be renamed separately to make it easier to review.

### Fixed
- Rename `Family` to `Case`

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: alembic/versions/20750539a335_add_ticket_to_families.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 Base = declarative_base()
 
 
-class Family(Base):
+class Case(Base):
     __tablename__ = "family"
```
Comment (ChrOertlin) [inline alembic/versions/20750539a335_add_ticket_to_families.py:27]:
```suggestion
    __tablename__ = "case"
```
are we going to change the table name in the database as well?

File: alembic/versions/998be2e367cf_fix_mip_on_fastq_wgs_cases.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ class FamilySample(Base):
     mother_id = sa.Column(sa.ForeignKey("sample.id"))
     father_id = sa.Column(sa.ForeignKey("sample.id"))
 
-    family = orm.relationship("Family", backref="links")
+    family = orm.relationship("Case", backref="links")
```
Comment (ChrOertlin) [inline alembic/versions/998be2e367cf_fix_mip_on_fastq_wgs_cases.py:66]:
```suggestion
    case = orm.relationship("Case", backref="links")
```

File: cg/cli/upload/genotype.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def upload_genotypes(context: CGConfig, re_upload: bool, family_id: Optional[str
     if not family_id:
```
Comment (ChrOertlin) [inline cg/cli/upload/genotype.py:37]:
```suggestion
    if not case_id:
```
Are we catching these instances in this PR as well or out of scope?

File: cg/meta/report/report_api.py [hunk 4]
```diff
@@ -108,14 +108,14 @@ def render_delivery_report(self, report_data: dict) -> str:
...
+    def get_cases_without_delivery_report(self, pipeline: Pipeline) -> list[Case]:
         """Returns a list of cases that has been stored and need a delivery report."""
-        stored_cases: list[Family] = []
+        stored_cases: list[Case] = []
         analyses: Query = self.status_db.analyses_to_delivery_report(pipeline=pipeline)[
             :MAX_ITEMS_TO_RETRIEVE
         ]
         for analysis_obj in analyses:
-            case: Family = analysis_obj.family
+            case: Case = analysis_obj.family
```
Comment (ChrOertlin) [inline cg/meta/report/report_api.py:118]:
```suggestion
            case: Case = analysis_obj.case
```
I think we need to patch this accross all store models

File: cg/meta/upload/scout/mip_config_builder.py [hunk 5]
```diff
@@ -186,7 +186,7 @@ def is_family_case(load_config: ScoutLoadConfig) -> bool:
     def is_multi_sample_case(load_config: ScoutLoadConfig) -> bool:
         return len(load_config.samples) > 1
 
-    def run_madeline(self, family_obj: Family) -> Path:
+    def run_madeline(self, family_obj: Case) -> Path:
```
Comment (ChrOertlin) [inline cg/meta/upload/scout/mip_config_builder.py:189]:
```suggestion
    def run_madeline(self, case: Case) -> Path:
```

Comment (ChrOertlin) [reply cg/meta/upload/scout/mip_config_builder.py:189]: there are two more instances of this in the function; ignore if out of scope.

File: cg/server/admin.py [hunk 6]
```diff
@@ -244,7 +244,7 @@ class CollaborationView(BaseView):
 
 
 class FamilyView(BaseView):
```
Comment (ChrOertlin) [inline cg/server/admin.py:246]:
```suggestion
class CaseView(BaseView):
```

File: cg/store/api/add.py [hunk 7]
```diff
@@ -226,7 +226,7 @@ def add_case(
 
     def relate_sample(
         self,
-        family: Family,
+        family: Case,
```
Comment (ChrOertlin) [inline cg/store/api/add.py:229]:
```suggestion
        case: Case,
```

File: cg/store/api/base.py [hunk 8]
```diff
@@ -90,18 +90,13 @@ def _get_join_sample_application_version_query(self) -> Query:
...
-        )
+        return self._get_query(table=Analysis).join(Case).join(Case.links).join(FamilySample.sample)
 
     def _get_subquery_with_latest_case_analysis_date(self) -> Query:
         """Return a subquery with the case internal id and the date of its latest analysis."""
         case_and_date: Query = (
             self._get_join_analysis_case_query()
-            .group_by(Family.id)
+            .group_by(Case.id)
             .with_entities(Analysis.family_id, func.max(Analysis.started_at).label("started_at"))
```
Comment (ChrOertlin) [inline cg/store/api/base.py:100]:
```suggestion
            .with_entities(Analysis.case_id, func.max(Analysis.started_at).label("started_at"))
```

File: cg/store/api/base.py [hunk 9]
```diff
@@ -180,18 +175,18 @@ def _get_filtered_case_query(
...
-            cases_query = cases_query.join(Family.links).join(FamilySample.sample)
+            cases_query = cases_query.join(Case.links).join(FamilySample.sample)
             cases_query = apply_sample_filter(
                 samples=cases_query,
                 filter_functions=[SampleFilter.FILTER_BY_INTERNAL_ID_PATTERN],
                 internal_id_pattern=sample_id,
             )
         else:
-            cases_query = cases_query.outerjoin(Family.links).outerjoin(FamilySample.sample)
+            cases_query = cases_query.outerjoin(Case.links).outerjoin(FamilySample.sample)
```
Comment (ChrOertlin) [inline cg/store/api/base.py:185]:
```suggestion
            cases_query = cases_query.outerjoin(Case.links).outerjoin(CaseSample.sample)
```

File: cg/store/api/find_business_data.py [hunk 10]
```diff
@@ -219,17 +219,17 @@ def get_cases_by_customer_pipeline_and_case_search(
...
         )
         return filtered_cases.limit(limit=limit).all()
 
-    def get_cases(self) -> list[Family]:
+    def get_cases(self) -> list[Case]:
         """Return all cases."""
-        return self._get_query(table=Family).all()
+        return self._get_query(table=Case).all()
 
     def get_case_samples_by_case_id(self, case_internal_id: str) -> list[FamilySample]:
```
Comment (ChrOertlin) [inline cg/store/api/find_business_data.py:234]:
```suggestion
    def get_case_samples_by_case_id(self, case_internal_id: str) -> list[CaseSample]:
```

File: cg/store/api/status.py [hunk 11]
```diff
@@ -111,7 +111,7 @@ def cases_to_analyze(
             pipeline=pipeline,
         )
 
-        families: list[Query] = list(cases.order_by(Family.ordered_at))
+        families: list[Query] = list(cases.order_by(Case.ordered_at))
```
Comment (ChrOertlin) [inline cg/store/api/status.py:114]:
```suggestion
        cases: list[Query] = list(cases.order_by(Case.ordered_at))
```

File: cg/store/models.py [hunk 12]
```diff
@@ -219,7 +219,7 @@ class Analysis(Model):
     family_id = Column(ForeignKey("family.id", ondelete="CASCADE"), nullable=False)
     uploaded_to_vogue_at = Column(types.DateTime, nullable=True)
 
-    family = orm.relationship("Family", back_populates="analyses")
+    family = orm.relationship("Case", back_populates="analyses")
```
Comment (ChrOertlin) [inline cg/store/models.py:222]:
```suggestion
    case = orm.relationship("Case", back_populates="analyses")
```

File: cg/store/models.py [hunk 13]
```diff
@@ -376,7 +376,7 @@ def to_dict(self):
         return to_dict(model_instance=self)
 
 
-class Family(Model, PriorityMixin):
+class Case(Model, PriorityMixin):
     __tablename__ = "family"
```
Comment (ChrOertlin) [inline cg/store/models.py:380]:
```suggestion
    __tablename__ = "case"
```

File: cg/store/models.py [hunk 14]
```diff
@@ -532,7 +532,7 @@ class FamilySample(Model):
     mother_id = Column(ForeignKey("sample.id"))
     father_id = Column(ForeignKey("sample.id"))
 
-    family = orm.relationship(Family, back_populates="links")
+    family = orm.relationship(Case, back_populates="links")
```
Comment (ChrOertlin) [inline cg/store/models.py:535]:
```suggestion
    case = orm.relationship(Case, back_populates="links")
```

**Review by ChrOertlin — APPROVED** [review]
Nice job! I think it looks good as is with renaming the model. However we should think about the other instances such as `FamilySample` and attributes in other models that refer to `family_id`, `family` etc. Should that be in separate prs?

What about variables that are still called `families`, `family_id` etc. Do we do that in separate PRs or in this one?

File: alembic/versions/20750539a335_add_ticket_to_families.py (hunk 1 above)
Comment (seallard) [reply alembic/versions/20750539a335_add_ticket_to_families.py:27]: Yes! But I'll leave it for a separate PR - might be possible to group the changes in a single revision

File: cg/meta/report/report_api.py (hunk 4 above)
Comment (seallard) [reply cg/meta/report/report_api.py:118]: Yeah, I'll save the renaming for those fields for a separate PR.

File: cg/server/app.py [hunk 15]
```diff
@@ -121,7 +121,7 @@ def _register_admin_views():
     )
 
     # Business data views
-    ext.admin.add_view(admin.FamilyView(Family, ext.db.session))
+    ext.admin.add_view(admin.FamilyView(Case, ext.db.session))
     ext.admin.add_view(admin.FamilySampleView(FamilySample, ext.db.session))
```
[the remark this answers is not in the document]
Comment (seallard) [reply cg/server/app.py]: Leaving the FamilySample stuff

File: cg/store/models.py (hunk 12 above)
Comment (seallard) [reply cg/store/models.py:222]: Yeah, leaving it for a separate PR though

**Pull request merged**